### PR TITLE
Python: add burned_mana field

### LINKED ISSUE #885: Add `burned_mana` field to Block - python
https://github.com/iotaledger/tips-draft/blob/tip46/tips/TIP-0046/tip-0046.md#basic-block

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Python: add burned_mana field

## Links to any relevant issues

Closes #885 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

File: bindings/python/iota_sdk/types/block.py [hunk 1]
```diff
@@ -18,11 +18,13 @@ class Block:
     Attributes:
         protocol_version: The protocol version with which this block was issued.
         parents: The parents of this block.
+        burned_mana: The amount of Mana the Account identified by the IssuerId is at most willing to burn for this block.
         payload: The optional payload of this block.
     """
 
     protocol_version: int
     parents: List[HexStr]
+    burned_mana: str
```
Comment (DaughterOfMars) [inline bindings/python/iota_sdk/types/block.py:27]: Not a number?

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/block.py:27]: No, it gets serialized as string

Comment (DaughterOfMars) [reply bindings/python/iota_sdk/types/block.py:27]: Yeah, but does that mean the field has to be a string in the type?

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/block.py:27]: Hm yeah if we add some custom dict/json methods then we should be able to do it, will have to investigate

Comment (DaughterOfMars) [reply bindings/python/iota_sdk/types/block.py:27]: Do we really use strings for all u64 in python?

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/block.py:27]: Currently we do

Comment (DaughterOfMars) [reply bindings/python/iota_sdk/types/block.py:27]: Dang, alright

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**